### ISSUE TITLE
Disable compression on Imgix images?

### ISSUE THREAD

**Issue by seaneking** [angeloashmore/gatsby-source-prismic#219]
Is there a way we can strip (or tweak) the `auto=compress` flag Prismic automatically adds to images images before handing them to the `gatsby-image` fragment? It causes huge issues with PNGs in Safari and other browsers ([example](https://images.prismic.io/bokeh/4c99cf14-3bde-4a4d-80d5-5cf1d7f8bb4f_Design.png?auto=compress%2Cformat&fit=max&w=1680&h=2192), view in Safari)

(I've contacted Prismic about it, their answer was "strip the URL flags" 🙃)

**Comment by angeloashmore** [comment]
Interesting, I haven't seen that issue in Safari before!

I'm working on `gatsby-plugin-imgix` to provide universal Imgix support which this plugin will leverage. Things like arbitrary Imgix transformations and quality settings will be configurable via `pluginOptions` and directly on fields as necessary.

The repo is private for now but an initial version and readme is published to NPM: https://www.npmjs.com/package/gatsby-plugin-imgix

**Comment by seaneking** [comment]
Amazing! What would the config look like for Prismic? Assume we don't get an imgix token from them?

Re: stripping default params, wouldn't that need to be done in the transformer since it's given to the fluid fragment to create a srcset? Unless I'm missing something really obvious. I haven't written sources or transformers before so I'm pretty fuzzy on how they actually work

**Comment by angeloashmore** [comment]
Plugin options could look something like the following:

```js
{
  resolve: 'gatsby-source-prismic',
  options: {
    // Other config here...

    // Override the default Imgix URL parameters.
    defaultImgixParams: {
      // Default: 'compress,format'
      auto: 'format',

      // Quality (0-100). Default: 50
      q: 100
    }

    defaultPlaceholderImgixParams: {
      // Width. Default: 100
      w: 30

      // Quality (0-100). Default: 20
      q: 10
    }
  }
}
```

And if the params need to be overridden on just a specific field, arguments could be passed to the field in the GraphQL query:

```gql
{
  fluid(
    imgixParams: { auto: 'format', q: 100 },
    placeholderImgixParams: { w: 30, q: 10 }
  )
}
```

That's the idea, at least. Let me know if you have any ideas on improving this. 👍

---

**Stripping `compress`** - The source plugin just returns an object so the stripping would take place where you're using the fixed/fluid object. I was thinking something like this:

```jsx
// Goes through each property and parses the URL to remove/replace the `auto` param.

<GatsbyImage fluid={patchImgixGatsbyImage(page.data.imageField.fluid)} />
```

If you don't need to support IE 11, `URLSearchParams` is a pretty handy way to edit the params.

**Comment by seaneking** [comment]
Nice! For the plugin opts, why not just provide a `imgix` and `imgixPlaceholder` objects with Prismic's defaults?

**Comment by angeloashmore** [comment]
This is part of the just published `v3.1.0`!

`imageImgixParams` and `imagePlaceholderImgixParams` are available as plugin options. The examples listed in the README's sample `gatsby-config.js` are the defaults.

To remove compression, set auto to one of the following:

```js
{
  imageImgixParams: {
    // Just remove automatic compression
    auto: 'format'

    // Remove the auto parameter completely
    auto: undefined
  }
}
```
